<a href="https://colab.research.google.com/github/walkerjian/Physics/blob/main/Two_hours_to_orbit_a_rock.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# It takes 2 hours to orbit 'at the surface' of any object made of rock!

I encountered a delightful little astrodynamics proof last week when an astronaut casually stated it as a fact, and when I looked skeptical, he just smiled and said, "Check it yourself." Here's the statement: "It takes 2 hours to orbit at the surface of any object made of rock"

To evaluate the statement "It takes 2 hours to orbit at the surface of any object made of rock," we'll need to derive the orbital period for a satellite orbiting just above the surface of a spherical object made of rock. This involves some basic astrodynamics principles.

We'll assume:
- The object is a perfect sphere.
- The object is homogeneous and made of rock, with a typical rock density.

Here's the general plan:
1. Calculate the mass of the object using its density and volume.
2. Use the mass to find the gravitational parameter.
3. Determine the orbital radius (which is the same as the object's radius).
4. Use Kepler's third law to calculate the orbital period.
5. Check if the orbital period is approximately 2 hours.

Let's start by defining the density of rock. The average density of rock is approximately 3000 kg/m³.

We'll write the Python code to perform these calculations:
This code will calculate the orbital period for three different radii: the radius of Earth, the Moon, and a hypothetical object with a radius of 1,000 km. You can add or modify radii as needed to test other objects.

The expected output will help us determine whether the statement "It takes 2 hours to orbit at the surface of any object made of rock" is true or not. If all calculated periods are approximately 2 hours, the statement holds; otherwise, it doesn't.

In [1]:
import math

# Constants
density_rock = 3000  # Density of rock in kg/m^3
G = 6.67430e-11  # Gravitational constant in m^3 kg^-1 s^-2

def calculate_orbital_period(radius):
    # Calculate the mass of the spherical object
    volume = (4/3) * math.pi * radius**3
    mass = density_rock * volume

    # Calculate the gravitational parameter (mu)
    mu = G * mass

    # Calculate the orbital period using Kepler's third law
    # T = 2 * pi * sqrt(r^3 / mu)
    period = 2 * math.pi * math.sqrt(radius**3 / mu)

    return period

# Test for different radii (e.g., radius of Earth, Moon, and other rocky objects)
radii = [6371000, 1737100, 1000000]  # Radii in meters (Earth, Moon, and a hypothetical 1,000 km radius object)
periods = [calculate_orbital_period(radius) for radius in radii]

# Convert periods from seconds to hours
periods_hours = [period / 3600 for period in periods]

# Print the results
for radius, period in zip(radii, periods_hours):
    print(f"Radius: {radius/1000} km, Orbital Period: {period:.2f} hours")


Radius: 6371.0 km, Orbital Period: 1.91 hours
Radius: 1737.1 km, Orbital Period: 1.91 hours
Radius: 1000.0 km, Orbital Period: 1.91 hours


Great! The results indicate that the orbital period for objects with different radii but made of rock is approximately 1.91 hours, which is close to the 2-hour mark stated in the original statement.

Let's delve a bit deeper into why this happens and generalize the calculations to see if the orbital period is consistently around 2 hours for any rocky object.

### Explanation
1. **Volume and Mass**:
   The mass $M$ of a spherical object is calculated using its volume $V$ and density $\rho$:
   $$
   V = \frac{4}{3} \pi R^3
   $$
   $$
   M = \rho V = \rho \left( \frac{4}{3} \pi R^3 \right)
   $$


2. **Gravitational Parameter**:
   The gravitational parameter $\mu$ (standard gravitational parameter) is:
   $$
   \mu = G M = G \rho \left( \frac{4}{3} \pi R^3 \right)
   $$


3. **Orbital Period**:
   Using Kepler's third law, the orbital period $T$ for a circular orbit at the surface of the sphere (orbital radius $R$) is:

   $$
   T = 2 \pi \sqrt{\frac{R^3}{\mu}} = 2 \pi \sqrt{\frac{R^3}{G \rho \left( \frac{4}{3} \pi R^3 \right)}} = 2 \pi \sqrt{\frac{1}{G \rho \left( \frac{4}{3} \pi \right)}}
   $$

   Notice that $R$ cancels out, meaning the orbital period $T$ depends only on the density $\rho$ and gravitational constant $G$, not on the radius $R$.


4. **Simplified Formula**:
   Let's simplify the formula:

   $$
   T = 2 \pi \sqrt{\frac{3}{4 \pi G \rho}}
   $$

   For $\rho = 3000 \, \text{kg/m}^3$:

   $$
   T \approx 2 \pi \sqrt{\frac{3}{4 \pi (6.67430 \times 10^{-11} \, \text{m}^3 \text{kg}^{-1} \text{s}^{-2}) (3000 \, \text{kg/m}^3)}}
   $$

Let's compute this to verify if it's close to 2 hours.

In [2]:
# Constants
density_rock = 3000  # Density of rock in kg/m^3
G = 6.67430e-11  # Gravitational constant in m^3 kg^-1 s^-2
pi = math.pi

# Calculate the orbital period using the simplified formula
T = 2 * pi * math.sqrt(3 / (4 * pi * G * density_rock))

# Convert the period from seconds to hours
T_hours = T / 3600

T_hours

1.9057659962346944

In [3]:
print(f"Theoretical Orbital Period: {T_hours:.2f} hours")


Theoretical Orbital Period: 1.91 hours


Let's see the exact theoretical period:



This should confirm the result. From the earlier results and the theoretical calculation, it seems that the orbital period for an object made of rock is indeed approximately 1.91 hours, not exactly 2 hours, but very close. This demonstrates the astronaut's statement was quite accurate, though not perfectly precise.

In [4]:
import math

# Constants
density_rock = 3000  # Density of rock in kg/m^3
G = 6.67430e-11  # Gravitational constant in m^3 kg^-1 s^-2
pi = math.pi

# Calculate the orbital period using the simplified formula
T = 2 * pi * math.sqrt(3 / (4 * pi * G * density_rock))

# Convert the period from seconds to hours
T_hours = T / 3600

T_hours

1.9057659962346944